test/controllers/api/v2/content_credentials_controller_test.rb

```diff
@@ -69,5 +69,75 @@ def test_cert_content
       assert_response :success, @response.body
       assert_equal content, @cert.reload.content
     end
+
+    def test_create
+      test_key_name = "testkey"
+      content = "abc123"
+      temp_content_file = Tempfile.new(content)
+      temp_content_file.write(content)
+      temp_content_file.rewind
+      uploaded_file = Rack::Test::UploadedFile.new(temp_content_file.path, "text/plain")
+      post :create, params: { :organization_id => @organization.id, :name => test_key_name, :content_type => "gpg_key", :content => uploaded_file}
+      assert_response :success
+    end
+
+    def test_create_existing_name
+      content = "abc123"
+      temp_content_file = Tempfile.new(content)
+      temp_content_file.write(content)
+      temp_content_file.rewind
+      uploaded_file = Rack::Test::UploadedFile.new(temp_content_file.path, "text/plain")
+      post :create, params: { :organization_id => @organization.id, :name => @gpg_key.name, :content_type => "gpg_key", :content => uploaded_file }
+      assert_response 422, @response.body
+      response = JSON.parse(@response.body)
+      assert_equal response["displayMessage"], "Validation failed: Name has already been taken"
+    end
+
+    def test_create_invalid_content
```

Comment:
this is covered by https://github.com/Katello/katello/blob/master/spec/models/gpg_key_spec.rb#L32